src/EnergyPlus/SwimmingPool.cc

```diff
@@ -1056,23 +1056,22 @@ void UpdatePoolSourceValAvg(EnergyPlusData &state, bool &SwimmingPoolOn) // .TRU
     // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
     SwimmingPoolOn = false;
 
-    // If this was never allocated, then there are no radiant systems in this input file (just RETURN)
+    // If this was never allocated, then there are no swimming pools in this input file (just RETURN)
     for (int PoolNum = 1; PoolNum <= state.dataSwimmingPools->NumSwimmingPools; ++PoolNum) {
         if (!allocated(state.dataSwimmingPools->Pool(PoolNum).QPoolSrcAvg)) return;
```

Comment:
QPoolSrcAvg shouldn't be a standalone array...right?  It should just be a plain Real64 sitting on the actual pool struct.  That would be one less array sitting out in state that has to be allocated, cleared, and checked.  I'd actually push to see that cleaned up here, or let me know why it shouldn't be that way.  While in there, I would encourage you to look for other things that are separately allocated but should just be member variables.  If you don't have time to do that for this bug fix, let me know and I could probably do it.